fplll/hlll.cpp

```diff
@@ -0,0 +1,636 @@
+/*
+   This file is part of fplll. fplll is free software: you
+   can redistribute it and/or modify it under the terms of the GNU Lesser
+   General Public License as published by the Free Software Foundation,
+   either version 2.1 of the License, or (at your option) any later version.
+
+   fplll is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with fplll. If not, see <http://www.gnu.org/licenses/>. */
+
+/* Template source file */
+
+#include "hlll.h"
+#include "util.h"
+
+FPLLL_BEGIN_NAMESPACE
+
+template <class ZT, class FT> bool HLLLReduction<ZT, FT>::hlll()
+{
+  /* TODO: we do not use a completely correct value for delta. We must use a value
```

Comment:
What is the status of this TODO.

Reply:
This TODO is related to the first limitation I listed in the comment of the PR. Indeed, in the paper that introduce [HLLL](http://perso.ens-lyon.fr/damien.stehle/downloads/HLLL.pdf), the delta to do the computation must be in an interval (delta_input + 2^(-p + p0), 1 - 2^(-p + p0)). I presume this is more or less the same in LLL, delta must be a bit modified to guarantee the computation. It is a TODO for a careful analysis about how to set delta to guarantee the HLLL-reduction w.r.t delta.